sphinx/directives/patches.py

```diff
@@ -55,9 +56,45 @@ def run(self):
         return result
 
 
+class RSTTable(tables.RSTTable):
+    """The table directive which sets source and line information to its caption."""
+
+    def make_title(self):
+        title, message = tables.RSTTable.make_title(self)
+        if title:
+            set_source_info(self, title)
+
+        return title, message
```

Comment:
I know that, but we can't use `apply_source_workaround()` in this case. It seems there are no way to obtain the line number of the table directives.... all of nodes related with table don't have it (table, thead, tbody and so on). So the workaround function could not fill the attributes.